.github/settings.yml

```diff
@@ -0,0 +1,22 @@
+repository:
+  # https://probot.github.io/apps/settings/
+  allow_merge_commit: false
+  allow_rebase_merge: true
+  allow_squash_merge: true
+  default_branch: main
+  delete_branch_on_merge: true
+  has_discussions: false
+  has_issues: true
+  has_projects: false
+  has_wiki: false
+  private: false
+branches:
+  - name: main
+    # https://docs.github.com/en/rest/reference/repos#update-branch-protection
+    protection:
```

Comment:
FYI:
We have discussed using `required_linear_history: true` with @bahadzie in the past but didn't come up with a satisfying solution.
The problem is that it will **not** remove the button "update branch" (implied: with a merge commit), but then block you from merging the PR is you clicked this button.